account-management-service/src/main/java/org/bf2/srs/fleetmanager/spi/impl/exception/AccountManagementSystemClientException.java

```diff
@@ -20,33 +25,52 @@
 
     public AccountManagementSystemClientException(Error causeEntity, int statusCode) {
         super(String.format("Error '%s' found when executing action. Returned status code is '%s'", causeEntity, statusCode));
+        this.causeException = Optional.empty();
         this.causeEntity = Optional.of(causeEntity);
         this.statusCode = Optional.of(statusCode);
     }
 
     public AccountManagementSystemClientException(String message, int statusCode) {
         super(message);
+        this.causeException = Optional.empty();
         this.causeEntity = Optional.empty();
         this.statusCode = Optional.of(statusCode);
     }
 
     public AccountManagementSystemClientException(String message) {
         super(message);
+        this.causeException = Optional.empty();
         this.causeEntity = Optional.empty();
         this.statusCode = Optional.empty();
     }
 
     public AccountManagementSystemClientException(Throwable error) {
         super(error.getMessage());
+        this.causeException = Optional.of(error);
         this.causeEntity = Optional.empty();
         this.statusCode = Optional.empty();
     }
 
-    public AccountManagementServiceClientException convert() {
+    private boolean shouldRetry() {
+        if (statusCode.isPresent()) {
+            return statusCode.get() / 100 == 5;
```

Comment:
Same here, I would extract the numbers to constants. For me, it makes it a bit harder to read the code.

Reply:
Makes sense, I'll update that

Reply:
I've added a clarifying comment instead, since I realized that the constant here is probably not needed.